tutorials/python/tutorial.md

```diff
@@ -0,0 +1,245 @@
+# Metaparticle/Package for Python Tutorial
+This is an in-depth tutorial for using Metaparticle/Package for Python
+
+For a quick summary, please see the [README](README.md).
+
+## Initial Setup
+
+### Check the tools
+The `docker` command line tool needs to be installed and working. Try:
+`docker ps` to verify this.  Go to the [install page](https://get.docker.io) if you need
+to install Docker.
+
+The `mp-compiler` command line tool needs to be installed and working.
+Try `mp-compiler --help` to verify this. Go to the [releases page](https://github.com/metaparticle-io/metaparticle-ast/releases) if you need to install
+the Metaparticle compiler.
+
+
+# Install Metaparticle/Package
+
+## Install the library
+```sh
+pip install metaparticle_pkg
+```
+
+## Get the tutorial code
+```sh
+git clone https://github.com/metaparticle-io/package
+cd package/tutorials/python
+```
+
+## Initial Program
+Inside of the `tutorials/python` directory, you will find a simple python project.
+
+```python
+from six.moves import SimpleHTTPServer, socketserver
+import socket
+
+OK = 200
+
+port = 8080
+
+class MyHandler(SimpleHTTPServer.SimpleHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(OK)
+        self.send_header("Content-type", "text/plain")
+        self.end_headers()
+        self.wfile.write("Hello Metparticle [{}] @ {}\n".format(self.path, socket.gethostname()).encode('UTF-8'))
+        print("request for {}".format(self.path))
+    def do_HEAD(self):
+        self.send_response(OK)
+        self.send_header("Content-type", "text/plain")
+        self.end_headers()
+
+def main():
+    Handler = MyHandler
+    httpd = socketserver.TCPServer(("", port), Handler)
+    httpd.serve_forever()
+
+if __name__ == '__main__':
+    main()
+```
+
+## Step One: Containerize the Application
+To build a container from our simple application we need to add a dependency to our
+code and then update the code to read as follows:
+
+```python
+from six.moves import SimpleHTTPServer, socketserver
```

Comment:
Maybe the README doesn’t need Python 2 compatibility stuff? Will keep examples clearer and end users are likely to be targeting one Python version. Maybe it is a good idea after all though to have an example that works with both.

Reply:
:+1: for Python 3, I don't expect anybody to use metaparticle for projects in legacy Python.